src/index.d.ts

```diff
@@ -128,7 +128,7 @@ export class MeshBVH {
         index2: number
       ) => boolean,
 
-      intersectsTriangles?: (
+      intersectsTriangles: (
```

Comment:
"intersectsTriangles" is technically optional. You can provide `intersectsRanges`, `intersectsTriangles`, or both callbacks.

Reply:
In theory yes, because you want to have the same behavior as the shapecast function, but the current implementation throws an error that `intersectsRanges` is not defined/not a function if `intersectsTriangles` is not provided, the default value of `intersectsRanges` is set inside the if null condition of `intersectsTriangles` few lines before.

Reply:
But calling the `bvhcast` while providing `intersectsRanges` but no `intersectsTriangles` callback is valid so it's not correct to say that `intersectsTriangles` is required. You have to provide one or the other, though.

Reply:
I tried to add a safeguard (with the _RequireAtLeastOne_ type) in the last commit, tell me what you think about that. Might be useful for the shapecast function as well. 
If you want to remove this and only "warn" in the doc, that's also fine to me.

Reply:
The require at least one approach seems fine to me. I'm not a typescript expert but the stackoverflow link is helpful. A more informative error could also be thrown if neither callback is provided but generally I don't feel strongly about that until I see it's causing more confusion.

It would be good to add it to shapecast, as well -- let me know if you want to add that otherwise this looks good to me and we can get it merged and published!